test/snippet/io/structure_file/structure_file_output.cpp

```diff
@@ -0,0 +1,254 @@
+};
+
+data_storage_t data_storage; // a global or globally used variable in your program
+//![data_storage]
+
+int main()
+{
+// Create a file /tmp/input.dbn for reading.
+{
+structure_file_out fout{"/tmp/input.dbn"};
+fout.emplace_back("GCGGAUUUAGCUCAGUUGGGAGAGCGCCAGACUGAAGAUUUGGAGGUCCUGUGUUCGAUCCACAGAAUUCGCA"_rna4,
+                  "S.cerevisiae_tRNA-PHE M10740/1-73",
+                  "(((((((..((((........)))).((((.........)))).....(((((.......))))))))))))."_wuss51);
+fout.emplace_back("UUGGAGUACACAACCUGUACACUCUUUC"_rna4,
+                  "example",
+                  "..(((((..(((...)))..)))))..."_wuss51);
+
+structure_file_out fout2{"/tmp/input_aa.dbn"};
+fout2.emplace_back("ACEWACEW"_aa20,
+                  "S.cerevisiae_tRNA-PHE M10740/1-73",
+                  "HGEBHHHH"_dssp9);
+fout2.emplace_back("ACEWACEWACEWACEW"_aa20,
+                  "example",
+                  "HGEBHHHHHGEBHHHH"_dssp9);
+}
+
+{
+//! [temp_param_deduc]
+structure_file_out fout{"/tmp/output.dbn"}; // Vienna format detected, std::ofstream opened for file
+//! [temp_param_deduc]
+}
+
+{
+//! [write_std_out]
+structure_file_out fout{std::cout, structure_file_format_vienna{}};
+//               ^ no need to specify the template arguments
+
+fout.emplace_back("AACGUU"_rna4, "example_id", ".(())."_wuss51); // default order for vienna: SEQ, ID, STRUCTURE
+//! [write_std_out]
+}
+
+{
+//! [iter_by_rec]
+structure_file_out fout{"/tmp/my.dbn"};
+
+for (int i = 0; i < 10; i++) // ...
+{
+    std::string id{"test_id"};
+    rna5_vector seq{"ACGU"_rna5};
+    std::vector<wuss51> structure{".()."_wuss51};
+
+    // ...
+
+    fout.emplace_back(seq, id, structure);        // as individual variables
+    // or:
+    fout.push_back(std::tie(seq, id, structure)); // as a tuple
+}
+//! [iter_by_rec]
+}
+
+// {
+// structured_rna<rna5, wuss51> temp{'G'_rna5, '.'_wuss51};
+// //! [write_fields]
+// structure_file_out fout{"/tmp/my.dbn", fields<field::ID, field::STRUCTURED_SEQ>{}};
+//
+// for (int i = 0; i < 10; i++)// ...
+// {
+//     std::string id{"test_id"};
+//     std::vector<structured_rna<rna5, wuss51>> structured_seq{temp, temp, temp, temp}; // vector of combined data structure
+//
+//     // ...
+//
+//     fout.emplace_back(id, structured_seq); // note also that the order the arguments is now different, because
+//     // or:                                    you specified that ID should be first in the fields template argument
+//     fout.push_back(std::tie(id, structured_seq));
+// }
+// //! [write_fields]
+// }
+
+// {
+// bool criteria = true;
+// //! [pass_rec]
+// structure_file_in  fin{"/tmp/input.dbn", fields<field::ID, field::SEQ, field::STRUCTURE>{}};
+// structure_file_out fout{"/tmp/my.dbn"}; // doesn't have to match the configuration
+//
+// for (auto & r : fin)
+// {
+//     if (criteria)// r fulfills some criterium
+//     {
+//         fout.push_back(r);
+//     }
+// }
+// //! [pass_rec]
+// }
```

Comment:
whats this?

Reply:
This just caught a fat bug.. Structure file was missing the push_back overload for records (not mere tuples) so the order fields was not correctly parsed.